src/api/cpp/cvc5.cpp

```diff
@@ -5027,15 +5027,21 @@ Proof::Proof(const std::shared_ptr<internal::ProofNode> p) : d_proof_node(p) {}
 
 Proof::~Proof()
 {
-  Assert(d_proof_node != nullptr);
-  d_proof_node.reset();
+  if (d_proof_node != nullptr)
```

Comment:
You don't need this condition.  (You actually likely don't need this entire block of code).

Reply:
Just make it empty.